args.go

```diff
@@ -544,13 +544,22 @@ func (s *argsScanner) next(kv *argsKV) bool {
 }
 
 func decodeArgAppend(dst, src []byte) []byte {
-	if bytes.IndexByte(src, '%') < 0 && bytes.IndexByte(src, '+') < 0 {
+	idxPercent := bytes.IndexByte(src, '%')
+	idxPlus := bytes.IndexByte(src, '+')
+	if idxPercent < 0 && idxPlus < 0 {
 		// fast path: src doesn't contain encoded chars
 		return append(dst, src...)
 	}
 
+	idx := min(idxPercent, idxPlus)
+	if idx > 0 {
```

Comment:
If one of them is -1 you can still start at the other. So I would change this block to be smarter without the min func.

Reply:
thx,   it  would  be better